<a href="https://colab.research.google.com/github/kylarisa/Innop_Less/blob/main/KPop2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
KPop=pd.read_csv("kpop_idol_followers.csv")
KPop.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 406 entries, 0 to 405
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Stage.Name      406 non-null    object 
 1   Group           406 non-null    object 
 2   ig_name         406 non-null    object 
 3   Followers       406 non-null    int64  
 4   Gender.x        406 non-null    object 
 5   Full.Name       406 non-null    object 
 6   Korean.Name     406 non-null    object 
 7   K.Stage.Name    406 non-null    object 
 8   Date.of.Birth   406 non-null    object 
 9   Debut           406 non-null    object 
 10  Company         406 non-null    object 
 11  Country         406 non-null    object 
 12  Second.Country  15 non-null     object 
 13  Height          248 non-null    float64
 14  Weight          175 non-null    float64
 15  Birthplace      288 non-null    object 
 16  Other.Group     79 non-null     object 
 17  Former.Group    35 non-null     obj

In [6]:
nKPop=KPop.isnull()
KPop.isnull().loc[(nKPop['Second.Country']==True)|(nKPop['Height']==True)|(nKPop['Weight']==True)|(nKPop['Birthplace']==True)|(nKPop['Other.Group']==True)|(nKPop['Former.Group']==True)]

Stage.Name  Group  ig_name  Followers  Gender.x  Full.Name  Korean.Name  \
0         False  False    False      False     False      False        False   
1         False  False    False      False     False      False        False   
2         False  False    False      False     False      False        False   
3         False  False    False      False     False      False        False   
4         False  False    False      False     False      False        False   
..          ...    ...      ...        ...       ...        ...          ...   
401       False  False    False      False     False      False        False   
402       False  False    False      False     False      False        False   
403       False  False    False      False     False      False        False   
404       False  False    False      False     False      False        False   
405       False  False    False      False     False      False        False   

     K.Stage.Name  Date.of.Birth  Debut  ...  Country  Second.Country  Height  \
0           False          False  False  ...    False            True   False   
1           False          False  False  ...    False            True    True   
2           False          False  False  ...    False            True    True   
3           False          False  False  ...    False            True   False   
4           False          False  False  ...    False           False   False   
..            ...            ...    ...  ...      ...             ...     ...   
401         False          False  False  ...    False            True   False   
402         False          False  False  ...    False            True   False   
403         False          False  False  ...    False            True   False   
404         False          False  False  ...    False            True    True   
405         False          False  False  ...    False            True    True   

     Weight  Birthplace  Other.Group  Former.Group  Gender.y    age  \
0     False        True         True          True     False  False   
1      True       False         True          True     False  False   
2      True       False         True          True     False  False   
3     False       False         True          True     False  False   
4      True       False         True          True     False  False   
..      ...         ...          ...           ...       ...    ...   
401   False       False         True          True     False  False   
402   False        True         True          True     False  False   
403   False       False        False          True     False  False   
404    True        True         True          True     False  False   
405    True       False         True          True     False  False   

     year.career  
0          False  
1          False  
2          False  
3          False  
4          False  
..           ...  
401        False  
402        False  
403        False  
404        False  
405        False  

[406 rows x 21 columns]

При вборе всех строк с нулевыми значениями встречающимися во всех возмодных столбцах, получается 406 строк, т.е. все строки датафрейма, При выборе всех строк с нулевыми значениями по нескольким из возможных столбзов, кодичество выбранныз строк меньше

In [7]:
KPop.corr()

<ipython-input-7-fdc5a3704d1f>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  KPop.corr()


Followers    Height    Weight       age  year.career
Followers     1.000000  0.082225  0.053297  0.002038     0.015670
Height        0.082225  1.000000  0.873428  0.306938     0.440308
Weight        0.053297  0.873428  1.000000  0.281591     0.406281
age           0.002038  0.306938  0.281591  1.000000     0.810746
year.career   0.015670  0.440308  0.406281  0.810746     1.000000

In [11]:
KPop['Followers'].corr(KPop['age'])

0.0020379345395163755

Наибольшая корреляция между возрастом 'age' и продолжительностью карьеры 'year.career'. Наименьшая между числом подписчиков 'Followers' и возрастом 'age'

In [12]:
KPop['Followers'].sum()

1477318792

In [13]:
KPop['Followers'].mean()

3638716.236453202

In [14]:
KPop['Followers'].median()

522469.0

In [15]:
KPop['Followers'].std()

9557984.574733391

In [16]:
KPop['Followers'].min()

718

In [17]:
KPop['Followers'].max()

93337766

In [19]:
print(nKPop!=KPop)

     Stage.Name  Group  ig_name  Followers  Gender.x  Full.Name  Korean.Name  \
0          True   True     True       True      True       True         True   
1          True   True     True       True      True       True         True   
2          True   True     True       True      True       True         True   
3          True   True     True       True      True       True         True   
4          True   True     True       True      True       True         True   
..          ...    ...      ...        ...       ...        ...          ...   
401        True   True     True       True      True       True         True   
402        True   True     True       True      True       True         True   
403        True   True     True       True      True       True         True   
404        True   True     True       True      True       True         True   
405        True   True     True       True      True       True         True   

     K.Stage.Name  Date.of.Birth  Debut